src/hierarchies/lin_reg_uni_hierarchy.cc

```diff
@@ -93,16 +99,16 @@ void LinRegUniHierarchy::draw() {
 void LinRegUniHierarchy::sample_given_data() {
   // Update values
   Hyperparams params = normal_invgamma_update();
-
   // Generate new state values from their prior centering distribution
   auto &rng = bayesmix::Rng::Instance().get();
   state.var = stan::math::inv_gamma_rng(params.shape, params.scale, rng);
   state.regression_coeffs = stan::math::multi_normal_prec_rng(
       params.mean, params.var_scaling / state.var, rng);
 }
 
-void LinRegUniHierarchy::sample_given_data(const Eigen::MatrixXd &data,
-                                           const Eigen::MatrixXd &covariates) {
+void LinRegUniHierarchy::sample_given_data(
```

Comment:
since this is a method mainly used for debugging (and in the SemiHDP), we could of having a less efficient version of it only in the BaseHierarchy. That is, this method clears the data already present, cycles through the input data & covariates calling `add_datum` and then calls the original `sample_given_data`